src/settings.rs

```diff
@@ -385,17 +385,41 @@ impl Settings {
         settings
     }
 
+    // Use ~/.mcfly only if it already exists, otherwise create 'mcfly' folder in XDG_CACHE_DIR 
     pub fn mcfly_training_cache_path() -> PathBuf {
-        Settings::storage_dir_path().join(PathBuf::from("training-cache.v1.csv"))
+        let cache_dir = Settings::mcfly_xdg_dir().cache_dir().to_path_buf();
+    
+        Settings::mcfly_base_path(cache_dir).join(PathBuf::from("training-cache.v1.csv"))
     }
-
-    pub fn storage_dir_path() -> PathBuf {
-        home_dir()
-            .unwrap_or_else(|| panic!("McFly error: Unable to access home directory"))
-            .join(PathBuf::from(".mcfly"))
-    }
-
+    
+    // Use ~/.mcfly only if it already exists, otherwise create 'mcfly' folder in XDG_DATA_DIR 
     pub fn mcfly_db_path() -> PathBuf {
-        Settings::storage_dir_path().join(PathBuf::from("history.db"))
+        let data_dir = Settings::mcfly_xdg_dir().data_dir().to_path_buf();
+    
+        Settings::mcfly_base_path(data_dir).join(PathBuf::from("history.db"))
+    }
+    
+    fn mcfly_xdg_dir() -> ProjectDirs {
+        ProjectDirs::from("io", "github",  "McFly").unwrap()
```

Comment:
What is the recommend standard for this?

Reply:
IIRC I’ve never seen such convention. `~/.local/share/mcfly` and `~/.cache/mcfly` would be fine and more standard.

Reply:
Just to clarify - there isn't a specific convention on the 3 three fields - qualifier, organization and project. As a matter of fact, qualifier and org names affect the folder names on MacOS and Windows, but not Linux. **That said these 2 fields can be empty, and I'll make them so**. In the long term, depending on the direction of this project, these fields can be updated.

Additional information and guidance on this is available [here](https://docs.rs/directories-next/2.0.0/directories_next/struct.ProjectDirs.html#method.from)